pkg/signal/testfiles/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"os"
+	"syscall"
+	"time"
+
+	"github.com/docker/docker/pkg/signal"
+)
+
+func main() {
+	sigmap := map[string]os.Signal{
+		"TERM": syscall.SIGTERM,
+		"QUIT": syscall.SIGQUIT,
+		"INT":  os.Interrupt,
+	}
+
+	defer time.Sleep(10 * time.Millisecond)
+
+	signal.Trap(func() {
+		time.Sleep(10 * time.Millisecond)
```

Comment:
Well there is a case in which if a `SIGTERM` or `SIGINT` is provided three times the process should quit directly without calling the `cleanup` function. So in case of `SIGTERM` we provide the signal multiple times, so we need the process to wait if we are providing the signal multiple times, otherwise the process would just quit and we wont be able to test that scenario.